packages/amplify-provider-awscloudformation/src/iterative-deployment/deployment-manager.ts

```diff
@@ -238,15 +312,10 @@ export class DeploymentManager {
     return gsis ? gsis.every(idx => idx.IndexStatus === 'ACTIVE') : true;
   };
 
-  private waitForIndices = async (stackParams: DeploymentMachineOp) => {
-    if (stackParams.tableNames.length) {
-      console.log('\nWaiting for DynamoDB table indices to be ready');
-      // cfn is async to ddb gsi creation the api can return true before the gsi creation starts
-      await new Promise(res => setTimeout(res, 2000));
-    }
+  private waitForActiveTables = async (tables: string[]): Promise<void> => {
+    if(tables.length) console.log(`Waiting for DynamoDB table indices to be ready`);
```

Comment:
nit linting and also use context.print()

Reply:
The Deployment Manager instance itself does not have access to the CLI context. We also console.error any failed pushes

Reply:
Moved into the spinner text.